imgui_widgets.cpp

```diff
@@ -3637,9 +3637,19 @@ static void    STB_TEXTEDIT_LAYOUTROW(StbTexteditRow* r, ImGuiInputTextState* ob
 }
 
 // When ImGuiInputTextFlags_Password is set, we don't want actions such as CTRL+Arrow to leak the fact that underlying data are blanks or separators.
-static bool is_separator(unsigned int c)                                        { return ImCharIsBlankW(c) || c==',' || c==';' || c=='(' || c==')' || c=='{' || c=='}' || c=='[' || c==']' || c=='|' || c=='\n' || c=='\r'; }
-static int  is_word_boundary_from_right(ImGuiInputTextState* obj, int idx)      { if (obj->Flags & ImGuiInputTextFlags_Password) return 0; return idx > 0 ? (is_separator(obj->TextW[idx - 1]) && !is_separator(obj->TextW[idx]) ) : 1; }
-static int  is_word_boundary_from_left(ImGuiInputTextState* obj, int idx)       { if (obj->Flags & ImGuiInputTextFlags_Password) return 0; return idx > 0 ? (!is_separator(obj->TextW[idx - 1]) && is_separator(obj->TextW[idx])) : 1; }
+static bool is_separator(unsigned int c)                                        { return ImCharIsBlankW(c) || c==',' || c==';' || c=='(' || c==')' || c=='{' || c=='}' || c=='[' || c==']' || c=='|' || c=='\n' || c=='\r' || c == '.'; }
+static int is_word_boundary_from_right(ImGuiInputTextState* obj, int idx)
+{
+    if (obj->Flags & ImGuiInputTextFlags_Password) return 0;
+    if (is_separator(obj->TextW[idx])) return 1;
```

Comment:
This line was added so that `ctrl+backspace` only deletes one character when the cursor is in front of a separator, which matches the behavior of nearly all text editors. Without this line, the buffer `cmd.|` will be fully cleared on `ctrl+backspace`, where with this line only the period will be deleted.

Reply:
@**restore deleted**